src/sortStudents.ts

```diff
@@ -1,16 +1,72 @@
-
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageGrade',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+function calculateAvarage(student: Student): number {
+  const sumGrades = student.grades.reduce((sum, grade) => sum + grade);
+  const avg = sumGrades / student.grades.length;
+
+  return avg;
+}
+
+type SortStudentsType = string | boolean | number;
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedStudents = [...students].sort((a, b) => {
+    let aVal: SortStudentsType;
+    let bVal: SortStudentsType;
+
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+        aVal = a[sortBy];
+        bVal = b[sortBy];
+        break;
+      case SortType.Age:
+        aVal = a.age;
+        bVal = b.age;
+        break;
+      case SortType.Married:
+        aVal = a.married;
+        bVal = b.married;
+        break;
```

Comment:
Use same aproach here
```suggestion
      case SortType.Age:
      case SortType.Married:
        aVal = a[sortBy];
        bVal = b[sortBy];
        break;
```